src/ilasm/asmman.cpp

```diff
@@ -828,8 +834,13 @@ void    AsmMan::EmitAssembly()
     WszMultiByteToWideChar(g_uCodePage,0,m_pAssembly->szName,-1,wzUniBuf,dwUniBuf);
 
     hr = m_pAsmEmitter->DefineAssembly(              // S_OK or error.
+#ifdef FEATURE_CORECLR
```

Comment:
This should not be ifdefed out for CoreCLR. It should be still possible to create assembly with a strong name, even on CoreCLR. Just the signing part - `Assembler::StrongNameSign` should be excluded.